internal/service/batch/job_definition_test.go

```diff
@@ -1680,6 +1779,186 @@ resource "aws_batch_job_definition" "test" {
 `, rName)
 }
 
+func testAccJobDefinitionConfig_ECSProperties_basic(rName string) string {
+	return fmt.Sprintf(`
+data "aws_partition" "current" {}
+
+resource "aws_iam_role" "ecs_task_execution_role" {
+  name               = %[1]q
+  assume_role_policy = data.aws_iam_policy_document.assume_role_policy.json
+}
+
+data "aws_iam_policy_document" "assume_role_policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "ecs_task_execution_role_policy" {
+  role       = aws_iam_role.ecs_task_execution_role.name
+  policy_arn = "arn:${data.aws_partition.current.partition}:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+resource "aws_batch_job_definition" "test" {
+  name = %[1]q
+  type = "container"
+  platform_capabilities = ["FARGATE"]
+  ecs_properties {
+    task_properties {
+      execution_role_arn = aws_iam_role.ecs_task_execution_role.arn
+      containers {
```

Comment:
It seems a bit odd to call it `containers` (plural) when the block defines a single container.
Shouldn't we call it `container` instead ?

Reply:
I see. fix in https://github.com/hashicorp/terraform-provider-aws/pull/37871/commits/10a4475ef5711adc94412978f3fc428e4f641758